thunder/clang/__init__.py

```diff
@@ -715,8 +731,49 @@ def _advanced_indexing(a: TensorLike, /, key) -> TensorLike:
         lambda: f"Advanced indexing currently only supports keys that are ellipses, integer tensors or sequences, but got {key=}",
     )
 
+    def _to_tensorproxies(x: list, device: devices.DeviceType):
+        # convert list to tensor
+        # e.g. [1, 2] -> tensor.Tensor([1, 2])
+        for idx, val in enumerate(x):
+            if isinstance(val, (NumberProxy)):
+                x[idx] = utils.get_numberlike_value(val)
+        if all(isinstance(i, int) for i in x):
+            x = tensor_from_sequence(x, dtype=dtypes.int64, device=device)
+        return x
+
+    basic_keys = []  # (key index, key)
+    advanced_keys = []  # (key index, key)
+
+    # convert list to 1D tensor
+    # this handles both cases when key is list or a tuple that contains lists
+    if isinstance(key, Sequence):
+        if isinstance(key, list):
```

Comment:
If we wanted to improve this, we could swap the if and else to condition on key being a TensorLike.

Reply:
I'm a little confused by the cases here -- maybe we could go through them in a comment? It seems like there should be some way to simplify the case logic.

Also, for this check (as currently written) isn't it any non-tuple sequence that we'd like to identify?

Reply:
What other sequences do we want to cover in advanced indexing? (Slices are covered in basic indexing)